argoverse/utils/make_att_files.py

```diff
@@ -0,0 +1,365 @@
+    """
+    image_size = 2000
+    image_scale = 10
+    img = np.zeros((image_size, image_size, 3))
+    pc = pc * image_scale
+    pc[:, 0] += int(image_size / 2)
+    pc[:, 1] += int(image_size / 2)
+    pc = pc.astype("int")
+
+    ind_valid = np.logical_and(
+        np.logical_and(pc[:, 0] >= 0, pc[:, 1] >= 0), np.logical_and(pc[:, 0] < image_size, pc[:, 1] < image_size)
+    )
+    img[pc[ind_valid, 0], pc[ind_valid, 1], :] = 0.4
+
+    path_imgs = os.path.join(path_output_vis, "bev")
+    if not os.path.exists(path_imgs):
+        os.mkdir(path_imgs)
+
+    for bbox, difficulty_att in zip(list_bboxes, list_difficulty_att):
+
+        qw, qx, qy, qz = bbox["rotation"]["w"], bbox["rotation"]["x"], bbox["rotation"]["y"], bbox["rotation"]["z"]
+        theta_local = np.arctan2(2 * (qw * qz + qx * qy), 1 - 2 * (qy * qy + qz * qz))
+        pose_local = np.array([bbox["center"]["x"], bbox["center"]["y"], bbox["center"]["z"]])
+
+        color_0 = 0.0
+        color_1 = 0.0
+        color_2 = 0.0
+        for att in difficulty_att:
+            color_0 += dict_color[att][0] / len(difficulty_att)
+            color_1 += dict_color[att][1] / len(difficulty_att)
+            color_2 += dict_color[att][2] / len(difficulty_att)
+        color = (color_0, color_1, color_2)
+
+        w, l, h = bbox["width"], bbox["length"], bbox["height"]
+        bbox_2d = np.array([[-l / 2, -w / 2, 0], [l / 2, -w / 2, 0], [-l / 2, w / 2, 0], [l / 2, w / 2, 0]])
+        R = np.array(
+            [[np.cos(theta_local), -np.sin(theta_local), 0], [np.sin(theta_local), np.cos(theta_local), 0], [0, 0, 1]]
+        )
+        bbox_2d = np.matmul(R, bbox_2d.transpose()).transpose() + pose_local[0:3]
+        edge_2d = np.array([[0, 1], [0, 2], [2, 3], [1, 3]])
+
+        for ii in range(len(edge_2d)):
+            p1 = (
+                int(bbox_2d[edge_2d[ii][0], 1] * image_scale + image_size / 2),
+                int(bbox_2d[edge_2d[ii][0], 0] * image_scale + image_size / 2),
+            )
+            p2 = (
+                int(bbox_2d[edge_2d[ii][1], 1] * image_scale + image_size / 2),
+                int(bbox_2d[edge_2d[ii][1], 0] * image_scale + image_size / 2),
+            )
+            cv2.line(img, p1, p2, color=color)
+
+    kernel = np.ones((5, 5), np.float)
+    img = cv2.dilate(img, kernel, iterations=1)
+    cv2.putText(
+        img,
+        "%s_%s_%d" % (dataset_name, log_id, lidar_timestamp),
+        (100, image_size - 100),
+        font,
+        fontScale,
+        fontColor,
+        lineType,
+    )
+
+    offset = 0
+    for key in dict_color.keys():
+        cv2.putText(img, key, (100 + offset, image_size - 50), font, fontScale, dict_color[key], lineType)
+        offset += 150
+
+    print("Saving img: ", path_imgs)
+    cv2.imwrite(os.path.join(path_imgs, "%s_%s_%d.jpg" % (dataset_name, log_id, lidar_timestamp)), img * 255)
+
+
+def bspline_1d(x: np.array, y: np.array, s: float = 20.0, k: int = 3) -> np.array:
```

Comment:
can we add the shape of `x`, `y`? And can we add what `s` and `k` represent in the B-Spline computation?